grimlock-core/src/main/scala/commbank/grimlock/spark/Implicits.scala

```
@@ -990,6 +992,22 @@ case class NativeOperations[X](data: Context.U[X]) extends FwNativeOperations[X,
 
     data.map(f)
   }
+
+  def take(num: Int)(implicit enc: Context.D[X]): Context.U[X] = {
+    import enc.ct
+
+    // Have to use `Some` to force the type to be non-primitive (Kryo cannot serialise Scala primitive types)
+    implicit val kryo: SparkEncoder[Some[X]] = Encoders.kryo[Some[X]]
+
+    SparkSession
+      .builder()
+      .config(data.sparkContext.getConf)
+      .getOrCreate()
+      .createDataset(data.map(Some.apply))
+      .limit(math.max(0, num))
```

maybe enforce with an assertion?

---

OK. So you'd rather throw an Exception if `num < 0`?

---

Only because I'm assuming that Scala's `List` will throw an exception if `num < 0`. Let's mimic whatever Scala's `List.take` does.

---

Ok. Scala list just returns an empty List if `num < 0`, so I'll keep it as it is now